src/mesh/voxel_cloud.rs

```diff
@@ -0,0 +1,456 @@
+                                &(voxel_coords + neighbor_offset),
+                            );
+                        }
+                    }
+                }
+            }
+        }
+
+        voxel_cloud
+    }
+
+    /// Gets the state of a voxel defined in voxel coordinates relative to the
+    /// voxel block start.
+    pub fn voxel_at_relative_coords(&self, relative_coords: &Point3<i32>) -> Option<bool> {
+        match self.relative_three_dimensional_coordinate_to_one_dimensional(relative_coords) {
+            Some(index) => Some(self.voxel_map[index]),
+            None => None,
+        }
+    }
+
+    /// Gets the state of a voxel defined in absolute voxel coordinates
+    /// (relative to the voxel space origin).
+    pub fn voxel_at_absolute_coords(&self, absolute_coords: &Point3<i32>) -> Option<bool> {
+        match self.absolute_three_dimensional_coordinate_to_one_dimensional(absolute_coords) {
+            Some(index) => Some(self.voxel_map[index]),
+            None => None,
+        }
+    }
+
+    /// Gets the state of a voxel containing the input point defined in model
+    /// space coordinates.
+    #[allow(dead_code)]
+    pub fn get_cartesian(&self, point: &Point3<f32>) -> Option<bool> {
+        let voxel_coords = self.cartesian_to_absolute_voxel_coords(point);
+        self.voxel_at_absolute_coords(&voxel_coords)
+    }
+
+    /// Sets the state of a voxel defined in voxel coordinates relative to the
+    /// voxel block start.
+    pub fn set_voxel_at_relative_coords(&mut self, state: bool, relative_coords: &Point3<i32>) {
+        let index = self
+            .relative_three_dimensional_coordinate_to_one_dimensional(relative_coords)
+            .expect("Coordinates out of bounds");
+        self.voxel_map[cast_usize(index)] = state;
+    }
+
+    /// Sets the state of a voxel defined in absolute voxel coordinates
+    /// (relative to the voxel space origin).
+    pub fn set_voxel_at_absolute_coords(&mut self, state: bool, absolute_coords: &Point3<i32>) {
+        let index = self
+            .absolute_three_dimensional_coordinate_to_one_dimensional(absolute_coords)
+            .expect("Coordinates out of bounds");
+        self.voxel_map[cast_usize(index)] = state;
+    }
+
+    /// Sets the state of a voxel containing the input point defined in model
+    /// space coordinates.
+    pub fn set_voxel_at_cartesian_coords(&mut self, state: bool, point: &Point3<f32>) {
+        let voxel_coords = self.cartesian_to_absolute_voxel_coords(point);
+        self.set_voxel_at_absolute_coords(state, &voxel_coords)
+    }
+
+    /// Calculates a simple triangulated welded mesh from the current state of
+    /// the voxel cloud.
+    ///
+    /// For watertight meshes this creates both, outer and inner boundary mesh.
+    /// There is also a high risk of generating a non-manifold mesh if some
+    /// voxels touch only diagonally.
+    pub fn to_mesh(&self) -> Option<Mesh> {
```

Comment:
This is also a very good candidate for a snapshot test, don't you think?

Reply:
after struggling with failing tests I realized the snapshot testing won't work because this function uses `weld` and that uses `HashMap`, therefore the result is always in a different order :/

Reply:
I also reviewed the weld function and for now I don't see a solution without the `HashMap`

Reply:
Okay, let's skip it for now.